example/apps/test_security/tests/utils.py

```diff
@@ -33,3 +36,27 @@ def test_utils_serialize_and_deserialize_data_should_return_right_values(self):
         }
         assert_equal(serialize_data(data), serialized_data)
         assert_equal(deserialize_data(serialized_data), data)
+
+    @data_consumer(
+        (
+            ('test', 'test', 10),
+            ('long truncated value without newline', '…\n newline', 10),
+            ('long truncated value with\nnewline', '…\nnewline', 10),
+            ('long truncated value with\nnewline', '…\nine', 5),
+        )
+    )
+    def test_truncate_newline_from_left_should_truncate_text(self, original_text, truncated_text, length):
+        assert_equal(truncate_newline_from_left(original_text, length), truncated_text)
+
+    @override_settings(
+        SECURITY_LOG_STRING_OUTPUT_TRUNCATE_LENGTH=100,
+        SECURITY_LOG_STRING_OUTPUT_TRUNCATE_OFFSET=10
+    )
+    def test_log_string_io_should_be_truncated_according_to_settings(self):
+        log_string_io = LogStringIO(write_time=False)
+        for _ in range(20):
+            log_string_io.write((4*'a') + '\n')
+        assert_equal(log_string_io.getvalue(), 20 * (4*'a' + '\n'))
+        assert_equal(len(log_string_io.getvalue()), 100)
+        log_string_io.write('b')
+        assert_equal(log_string_io.getvalue(), '…\n' + 17 * (4 * 'a' + '\n') + 'b')
```

Comment:
Why `17` instead of `19`? The `b` causes exceeding the limit, so one line is stripped off and replaced with `…\n`. (i know tests are passing, maybe I am a little tired already a not getting it)

Reply:
Two reasons, offset 10 characters which will do 18 but, b character is computed to the limit to. This is the reason why 17